sentinelhub/sentinelhub_request.py

```diff
@@ -1,89 +1,201 @@
-""" Helper functions for generating request dictionaries
+""" SentinelHubRequest for the Processing API
 
-Request structure documentation is available at: https://docs.sentinel-hub.com/api/latest/reference/
+Documentation: https://docs.sentinel-hub.com/api/latest/reference/
 """
 
-from .geometry import Geometry
+from datetime import datetime
 
+from .config import SHConfig
+from .constants import MimeType, DataSource, RequestType
+from .download import DownloadRequest, SentinelHubDownloadClient
+from .data_request import DataRequest
+from .geometry import Geometry, BBox
 
-def body(request_bounds, request_data, evalscript, request_output=None):
-    """ Generate request body
+
+class SentinelHubRequest(DataRequest):
+    """ Sentinel Hub API request class
     """
-    request_body = {
-        "input": {
-            "bounds": request_bounds,
-            "data": request_data
-        },
-        "evalscript": evalscript
-    }
+    def __init__(self, evalscript, input_data, responses, bounds=None, size=None, resolution=None, maxcc=1.0,
+                 mosaicking_order='mostRecent', config=None, mime_type=MimeType.TIFF, **kwargs):
+        """
+        :param bounds: Bounding box or geometry object
+        :type bounds: sentinelhub.BBox or sentinelhub.Geometry
+        """
+
+        if size is None and resolution is None:
+            raise ValueError("Either size or resolution argument should be given")
+
+        if not isinstance(evalscript, str):
+            raise ValueError("'evalscript' should be a string")
+
+        self.config = config or SHConfig()
+
+        self.mime_type = mime_type
+
+        self.payload = self.body(
+            request_bounds=self.bounds(bounds),
+            request_data=input_data,
+            request_output=self.output(size=size, resolution=resolution, responses=responses),
+            evalscript=evalscript
+        )
+
+        super().__init__(SentinelHubDownloadClient, **kwargs)
+
+    def get_data(self, save_data=False, redownload=False, max_threads=None):
+        """ Overrides DataRequest.get_data
+        """
+        self.download_list = [self._download_request]
+        return super().get_data(save_data=save_data, redownload=redownload, max_threads=max_threads)[0]
+
+    def create_request(self):
+        """ Prepares a download request
+        """
+        headers = {'content-type': 'application/json', "accept": self.mime_type.get_string()}
```

Comment:
Instead of `'application/json'` you can use `MimeType.JSON.get_string()`

Reply:
Done.